packages/core/src/lib/list-box.svelte

```diff
@@ -0,0 +1,249 @@
+import { Icon } from '$lib';
+
+const dispatch = createEventDispatcher();
+
+// Toggle element on or off
+export let disabled = false;
+// Elements in the left list
+export let left = '';
+// Elements in the right list
+export let right = '';
+// Label for the left list
+export let leftlabel = '';
+// Label for the right list
+export let rightlabel = '';
+// Height of the two boxes
+export let height = '200px';
+// Enable suffix that can be put in front of elements in the box
+export let suffix = false;
+
+type ListBoxSide = 'left' | 'right';
+
+interface ListBoxOption {
+  value: string;
+  selected: boolean;
+  suffix?: string;
+}
+
+interface Options {
+  left: ListBoxOption[];
+  right: ListBoxOption[];
+}
+
+const LEFT = 'left';
+const RIGHT = 'right';
+
+const generateOption = (initial: string): ListBoxOption => {
+  if (suffix) {
+    const split = initial.split(' ');
+    return { value: split[0] ?? '', suffix: split[1] ?? '', selected: false };
+  }
+
+  return { value: initial, selected: false };
+};
+
+let options = {
+  left: left ? left.split(',').map((val) => generateOption(val)) : [],
+  right: right ? right.split(',').map((val) => generateOption(val)) : [],
+};
+
+// when new values are added, we want to update the state of options to include those values
+const addNewData = () => {
+  const allValues = new Set([
+    ...options.left.map((opt) => opt.value),
+    ...options.right.map((opt) => opt.value),
+  ]);
+  const newLeftOptions = left
+    ? left
+        .split(',')
+        .map((val) => generateOption(val))
+        .filter((opt) => !allValues.has(opt.value))
+    : [];
+  const newRightOptions = right
+    ? right
+        .split(',')
+        .map((val) => generateOption(val))
+        .filter((opt) => !allValues.has(opt.value))
+    : [];
+
+  const newOptions = {
+    left: [...options.left, ...newLeftOptions],
+    right: [...options.right, ...newRightOptions],
+  };
+
+  options = newOptions;
+};
+
+afterUpdate(addNewData);
+
+const handleOptionClick = (option: ListBoxOption, side: ListBoxSide) => {
+  if (disabled) {
+    return;
+  }
+
+  option.selected = !option.selected;
+  options = { ...options };
+
+  dispatch('option-click', {
+    target: { ...option, side },
+  });
+};
+
+const handleMoveClick = (target: ListBoxSide) => {
+  if (disabled) {
+    return;
+  }
+
+  const source = target === LEFT ? RIGHT : LEFT;
+
+  const newOptions: Options = {
+    left: [],
+    right: [],
+  };
+
+  for (const option of options[source]) {
+    if (option.selected) {
+      newOptions[target].push({ ...option, selected: false });
+    } else {
+      newOptions[source].push(option);
+    }
+  }
+
+  options[source] = newOptions[source];
+  options[target] = [...options[target], ...newOptions[target]];
+  options = { ...options };
+
+  // copy object to avoid any potential pass-by-reference issues
+  dispatch('move', {
+    options: JSON.parse(JSON.stringify(options)) as Options,
+  });
```

Comment:
All this looks... a little sus. `options` is internal to this component, what's all this spreading / JSON (de)serialization doing?

Reply:
I didnt write the original version of this component but from what I understand json deserialization is a way easy way to make a deep copy of the object.

Reply:
My question was more along the lines of "why are we doing this cloning?" Looking at this block:

- L137: mutates the `options` object
- L128: mutates the `options` object again
- L129: shallow-clones the `options` object
- L143: deep-clones the `options object and places it in an event

So the reason I say it's "sus" is that we're inconsistent with mutations and cloning, and it's an internal object. It all feels like a design problem with this component's API (props and events)